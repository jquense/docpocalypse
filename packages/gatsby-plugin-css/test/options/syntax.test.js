const { webpack } = require('../helpers');

describe('Options', () => {
  test('Syntax - {String}', () => {
    const config = {
      loader: {
        test: /\.sss$/,
        options: {
          syntax: 'sugarss',
        },
      },
    };

    return webpack('sss/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1];

      expect(source).toEqual('module.exports = "a\\n  color: black\\n"');
      expect(source).toMatchSnapshot();
    });
  });

  test('Syntax - {Object}', () => {
    const config = {
      loader: {
        test: /\.sss$/,
        options: {
          ident: 'postcss',
          syntax: require('sugarss'),
        },
      },
    };

    return webpack('sss/index.js', config).then((stats) => {
      const { source } = stats.toJson().modules[1];

      expect(source).toEqual('module.exports = "a\\n  color: black\\n"');
      expect(source).toMatchSnapshot();
    });
  });
});
