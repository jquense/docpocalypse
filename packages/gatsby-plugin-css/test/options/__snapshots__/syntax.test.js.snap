// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options Syntax - {Object} 1`] = `"module.exports = \\"a\\\\n  color: black\\\\n\\""`;

exports[`Options Syntax - {String} 1`] = `"module.exports = \\"a\\\\n  color: black\\\\n\\""`;
